pkg/migrations/sql/preinstall/000-utils.sql

```diff
@@ -1,9 +1,24 @@
-CREATE OR REPLACE PROCEDURE execute_everywhere(command TEXT, transactional BOOLEAN = true)
+ --perms for schema will be addressed later;
+ CREATE SCHEMA IF NOT EXISTS SCHEMA_CATALOG;
+
+--table to save commands so they can be run when adding new nodes
+ CREATE TABLE SCHEMA_CATALOG.remote_commands(
+    key TEXT PRIMARY KEY,
+    seq SERIAL,
+    transactional BOOLEAN,
+    command TEXT
+);
+
+
+CREATE OR REPLACE PROCEDURE execute_everywhere(command_key text, command TEXT, transactional BOOLEAN = true)
 AS $func$
 BEGIN
     EXECUTE command;
 
     BEGIN
+        INSERT INTO SCHEMA_CATALOG.remote_commands(key, command, transactional) VALUES(command_key, command, transactional)
+        ON CONFLICT (key) DO UPDATE SET command = excluded.command, transactional = excluded.transactional;
+
```

Comment:
I have a concern about non-idempotent commands, or things like temporarily creating tables, where executing them twice is the correct thing to do; can you expand a bit on why we do it like this instead of recording the command executed twice? The execute-multiple-times alternative _would_ require different functions for recording and replay.

Reply:
Oh wait, I misunderstood, the current one requires the user to create a unique key for each time `execute_everywhere()` is called. Can you comment on why that instead of just recording the sequence of commands we execute everywhere on the first go, then replaying all of them when we add a new node? (I could see it play _interestingly_ with update scripts, and with enough manual intervention the current version could be better...)